flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/HeapKeyedStateBackend.java

```diff
@@ -169,13 +169,16 @@ public HeapKeyedStateBackend(
 			TypeSerializer<N> namespaceSerializer,
 			TypeSerializer<V> valueSerializer) throws StateMigrationException {
 
-		final RegisteredKeyedBackendStateMetaInfo<N, V> newMetaInfo =
-				new RegisteredKeyedBackendStateMetaInfo<>(stateType, stateName, namespaceSerializer, valueSerializer);
-
 		@SuppressWarnings("unchecked")
 		StateTable<K, N, V> stateTable = (StateTable<K, N, V>) stateTables.get(stateName);
 
 		if (stateTable == null) {
+			RegisteredKeyedBackendStateMetaInfo<N, V> newMetaInfo = new RegisteredKeyedBackendStateMetaInfo<>(
```

Comment:
Mixing of concerns is not yet as bad as in the Rocks backend code, you might also start separating this a bit more here as well.